mypy.ini

```diff
@@ -0,0 +1,2 @@
+[mypy]
+python_version = 3.11
```

Comment:
I'd set this to 3.7, if you only do one, the lowest is best, usually.

Reply:
You can use those things regardless of the target version, at least in annotations. You might not be able to use them within type aliases - but it's much more "normal" to give up on type aliases and/or 

IMO, there's no harm at all adding `typing_extensions` to the requirements - any requirement that is only required on older Pythons is safe to add, as you aren't adding a requirement for latest and future Python users. So don't try to work around that, IMO.

I recommend adding `awkward.typing`, and doing all conditional imports for `typing`/`typing_extensions` there.

```python
BehaviorType: TypeAlias = "dict[str, ak._v2.highlevel.Array | ak._v2.record.Record]"
```

Is the "correct" way to do "nice" type aliases, I think.

Reply:
Yes, it's really me trying to avoid having lots of 
```python
`if sys.version(....):
    import ...
else:
    import ...
```

I think I'll just bite the bullet and add a typing submodule ;)

Reply:
In annotations, yes. It _might_ complain if they are used outside of future annotations. But in annotations they are perfectly valid - that's kind of the point of the future import. :)

Reply:
I thought that PEP563 just tackles deferred evaluation. Does mypy also use this as a cue to support the latest annotations? That would be useful!